src/http.c

```diff
@@ -190,29 +190,48 @@ static void requests_fail(um_http_t *c, int code) {
 }
 
 static void connect_cb(uv_connect_t *req, int status) {
-    um_http_t *c = req->data;
+    um_http_t *clt = req->data;
+    UM_LOG(VERB, "connected status = %d", status);
 
     if (status == 0) {
-        uv_link_read_start(&c->http_link);
+        uv_link_source_init(&clt->conn_src, (uv_stream_t *) &clt->conn);
+        uv_link_init(&clt->http_link, &http_methods);
+        clt->http_link.data = clt;
 
-        if (!c->ssl) {
-            c->connected = true;
-            uv_async_send(&c->proc);
+        if (clt->ssl) {
+            clt->tls_link = malloc(sizeof(uv_link_t));
```

Comment:
does this ever get free'd? i see it in here getting allocated but scanning the PR i don't see it being free'd. maybe it's on a previous pr but figured i'd ask

Reply:
changed it into a member struct (not pointer), no longer needs to be allocated/freed